torch/csrc/distributed/c10d/reducer.cpp

```diff
@@ -385,7 +391,10 @@ void Reducer::mark_variable_ready_sparse(VariableIndex index) {
     // struct are empty, and there is no pre-existing accumulation tensor.
     // Directly assign the sparse tensor to the `contents` field.
     replica.contents = grad;
-    replica.contents.div_(process_group_->getSize());
+    // See Note [DDP Communication Hook]
+    if (comm_hook_ == nullptr) {
+      replica.contents.div_(process_group_->getSize());
+    }
```

Comment:
We should probably test the communication hook with sparse tensors as well. Using [nn.EmbeddingBag](https://pytorch.org/docs/stable/generated/torch.nn.EmbeddingBag.html) with `sparse=True` will generate sparse gradients for you.

Reply:
@pritamdamania87 Thanks for this comment! It was quite useful to discover and fix a bug with sparse tensors case. 
I think we should just copy in case of sparse gradients, since `bucket_view` is not used. Please see `test_ddp_comm_hook_sparse_gradients` and changes in `reducer.cpp` to make it work.